src/dist/prefix.rs

```diff
@@ -2,7 +2,14 @@ use std::path::{Path, PathBuf};
 
 use crate::utils;
 
-const REL_MANIFEST_DIR: &str = "lib/rustlib";
+/// The relative path to the manifest directory in a Rust installation,
+/// with path components separated by [`std::path::MAIN_SEPARATOR`].
+const REL_MANIFEST_DIR: &str = match std::path::MAIN_SEPARATOR {
+    '/' => "lib/rustlib",
+    '\\' => r"lib\rustlib",
+    _ => panic!("unknown `std::path::MAIN_SEPARATOR`"),
+};
```

Comment:
I feel like we should be able to do this better but I couldn't come up with anything better than a macro that just wasn't worth it for one use. So on reflection this is fine.

Reply:
I've successfully implemented const-concating but that has been way too verbose for only this, so I truncated it to that.